### PR TITLE
More detailed explanation how do engines work in 3.0

### PR DESCRIPTION
**Pull request by beldmit**
Related: #16868, #17081, #17107

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by t8m** [comment]
I am not quite sure this is fully correct. Perhaps there are some bugs to be fixed. It does not seem to me as all these limitations are intentional.

**Comment by beldmit** [comment]
I am also not sure. I can mark this as draft but it's some information we must provide.

File: doc/man7/migration_guide.pod [hunk 1]
```diff
@@ -121,6 +121,27 @@ FIPS module, as detailed below. Authors and maintainers of external engines are
...
+New algorithms provided via engines will still work.
+
+Engine-backed keys can be loaded via custom B<OSSL_STORE> implementation.
+In this case the B<EVP_PKEY> objects created via L<ENGINE_load_private_key(3)>
+will be concidered legacy and will continue to work.
+
+Engine-based hardware acceleration for asymmetric algorithms will be available
+if you use the legacy API (e.g. B<RSA_METHOD>) and create the RSA object from
+parameters.  However, if you get the B<RSA> object with
+L<PEM_read_bio_PrivateKey(3)> and L<EVP_PKEY_get1_RSA(3)>, the hardware offload will
```
Comment (mattcaswell) [inline doc/man7/migration_guide.pod:138]: This is not correct. Please see my latest post in #17081

Comment (beldmit) [reply doc/man7/migration_guide.pod]: Fixed. Thanks for your investigation!

**Review by paulidale — APPROVED** [review]

**Comment by beldmit** [comment]
Merged. Thanks!

**Pull request closed without merging**